packages/fal/src/fal-transcription-model.ts

```diff
@@ -0,0 +1,380 @@
+   */
+  numSpeakers: z.number().nullable().nullish(),
+});
+
+export type FalTranscriptionCallOptions = z.infer<
+  typeof falProviderOptionsSchema
+>;
+
+interface FalTranscriptionModelConfig extends FalConfig {
+  _internal?: {
+    currentDate?: () => Date;
+  };
+}
+
+export class FalTranscriptionModel implements TranscriptionModelV1 {
+  readonly specificationVersion = 'v1';
+
+  get provider(): string {
+    return this.config.provider;
+  }
+
+  constructor(
+    readonly modelId: FalTranscriptionModelId,
+    private readonly config: FalTranscriptionModelConfig,
+  ) {}
+
+  private async getArgs({
+    providerOptions,
+  }: Parameters<TranscriptionModelV1['doGenerate']>[0]) {
+    const warnings: TranscriptionModelV1CallWarning[] = [];
+
+    // Parse provider options
+    const falOptions = parseProviderOptions({
+      provider: 'fal',
+      providerOptions,
+      schema: falProviderOptionsSchema,
+    });
+
+    // Create form data with base fields
+    const body: Omit<FalTranscriptionAPITypes, 'audio_url'> = {
+      task: 'transcribe',
+      diarize: true,
+      chunk_level: 'word',
+    };
+
+    // Add provider-specific options
+    if (falOptions) {
+      body.language = falOptions.language;
+      body.version = falOptions.version ?? undefined;
+      body.batch_size = falOptions.batchSize ?? undefined;
+      body.num_speakers = falOptions.numSpeakers ?? undefined;
+
+      if (typeof falOptions.diarize === 'boolean') {
+        body.diarize = falOptions.diarize;
+      }
+
+      if (falOptions.chunkLevel) {
+        body.chunk_level = falOptions.chunkLevel;
+      }
+    }
+
+    return {
+      body,
+      warnings,
+    };
+  }
+
+  async doGenerate(
+    options: Parameters<TranscriptionModelV1['doGenerate']>[0],
+  ): Promise<Awaited<ReturnType<TranscriptionModelV1['doGenerate']>>> {
+    const currentDate = this.config._internal?.currentDate?.() ?? new Date();
+    const { body, warnings } = await this.getArgs(options);
+
+    const base64Audio =
+      typeof options.audio === 'string'
+        ? options.audio
+        : convertUint8ArrayToBase64(options.audio);
+
+    const audioUrl = `data:${options.mediaType};base64,${base64Audio}`;
+
+    const { value: queueResponse } = await postJsonToApi({
+      url: this.config.url({
+        path: `https://queue.fal.run/fal-ai/${this.modelId}`,
+        modelId: this.modelId,
+      }),
+      headers: combineHeaders(this.config.headers(), options.headers),
+      body: {
+        ...body,
+        audio_url: audioUrl,
+      },
+      failedResponseHandler: falFailedResponseHandler,
+      successfulResponseHandler:
+        createJsonResponseHandler(falJobResponseSchema),
+      abortSignal: options.abortSignal,
+      fetch: this.config.fetch,
+    });
+
+    // Poll for completion with timeout
+    const startTime = Date.now();
+    const timeoutMs = 60000; // 60 seconds timeout
+    const pollIntervalMs = 1000; // 1 second interval
```

Comment:
wonder if this should be configurable in some way, maybe not initially

Reply:
Yeah agreed - we're starting to see the poll pattern in a few providers, maybe we can pull this out into a `poll` function of sorts and expose these config options to the user in a later PR.